spec/acceptance/agent_spec.rb

```diff
@@ -36,42 +36,53 @@ class { 'zabbix::agent':
     describe file('/etc/zabbix/zabbix_agentd.conf') do
       its(:content) { is_expected.not_to match %r{ListenIP=} }
     end
-  end
-end
 
-describe 'zabbix::agent class with zabbix_version 3.4' do
-  context 'With ListenIP set to an IP-Address' do
-    it 'works idempotently with no errors' do
-      pp = <<-EOS
-      class { 'zabbix::agent':
-        server               => '192.168.20.11',
-        zabbix_package_state => 'latest',
-        listenip             => '127.0.0.1',
-        zabbix_version       => '3.4'
-      }
-      EOS
-      apply_manifest(pp, catch_failures: true)
-      apply_manifest(pp, catch_changes: true)
-    end
-    describe file('/etc/zabbix/zabbix_agentd.conf') do
-      its(:content) { is_expected.to match %r{ListenIP=127.0.0.1} }
+    context 'With ListenIP set to an IP-Address' do
+      it 'works idempotently with no errors' do
+        pp = <<-EOS
+          class { 'zabbix::agent':
+            server               => '192.168.20.11',
+            zabbix_package_state => 'latest',
+            listenip             => '127.0.0.1',
+            zabbix_version       => '3.4'
```

Comment:
If you put this in the context and test it with every version then this should use the variable